src/java/src/main/java/com/dnanexus/DXWorkflow.java

```diff
@@ -239,4 +243,101 @@ public Describe getCachedDescribe() {
         return ExecutableRunner.getWorkflowRunnerWithEnvironment(this.getId(), this.env);
     }
 
+    /**
+     * A workflow stage
+     *
+     */
+    public static class DXStage {
+        private String ID;
+
+        DXStage(String ID) {
+            this.ID = ID;
+        }
+
+        public String getId() {
+            return ID;
+        }
+
+        /**
+         * Create a link to an output field. This is used in workflows, to
+         * link results between stages.
+         *
+         * @param fieldName  name of an output field
+         *
+         * @return JSON representation of a link. Can be used as an input to a workflow stage.
+         */
+        public ObjectNode getOutputReference(String  fieldName) {
+            ObjectNode dxlink = DXJSON.getObjectBuilder()
+                .put("stage", ID)
+                .put("outputField", fieldName).build();
+            return DXJSON.getObjectBuilder().put("$dnanexus_link", dxlink).build();
+        }
+
+        /**
+         * Create a link to an input field. This is used in workflows, to
+         * link results between stages.
+         *
+         * @param fieldName  name of an input field
+         *
+         * @return JSON representation of a link. Can be used as an input to a workflow stage.
+         */
+        public ObjectNode getInputReference(String  fieldName) {
+            ObjectNode dxlink = DXJSON.getObjectBuilder()
+                .put("stage", ID)
+                .put("inputField", fieldName).build();
+            return DXJSON.getObjectBuilder().put("$dnanexus_link", dxlink).build();
+        }
+    }
+
+    /** Some workflow update operations return an edit version. This class is a way
+     *  to report the new edit version, together with the operation result.
+     */
+    public static class Modification<T> {
+        public int editVersion;
```

Comment:
Please provide getters for these fields `getEditVersion`, `getValue` and make them `final`.